pkg/timeutil/periodic.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package timeutil
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+type periodicStart struct {
+	dayOfWeek      int
+	hourOfDay      int
+	minuteOfHour   int
+	secondOfMinute int
+}
+
+type Periodic struct {
+	start    *periodicStart
+	duration time.Duration
+}
+
+type Period struct {
+	Start time.Time
+	End   time.Time
+}
+
+// ParsePeriodic returns a Periodic specified as a start and duration.
+func ParsePeriodic(start, duration string) (*Periodic, error) {
+	var err error
+
+	pc := &Periodic{}
+
+	if pc.start, err = parseStart(start); err != nil {
+		return nil, fmt.Errorf("unable to parse start: %v", err)
+	}
+
+	if pc.duration, err = time.ParseDuration(duration); err != nil {
+		return nil, fmt.Errorf("unable to parse duration: %v", err)
+	}
+
+	if pc.duration < time.Duration(0) {
+		return nil, fmt.Errorf("duration cannot be negative")
+	}
+
+	// check that the duration of the window does not exceed the period.
+	if (pc.start.dayOfWeek == -1 && pc.duration > 24*time.Hour) || pc.duration > 7*24*time.Hour {
+		return nil, fmt.Errorf("duration cannot exceed period")
```

Comment:
hrm, wouldn't it make sense to have a duration equal to the period be treated as an error too?